docs/element-definitions.md

```diff
@@ -2169,12 +2169,13 @@ The HREF values for these transitions MAY be either relative or absolute URLs.
 
 - `element` - `"OAuth2 Scheme"`
 - `content` (array)
-	- ([Member][])
-	- ([Transition][])
+    - ([Array][])
```

Comment:
I wonder if it would be clearer to introduce a new element "OAuth2 Flow", for example:

+ element: OAuth2 Flow
+ attributes
	+ grantType
    + scopes
+ content (array[Member, Transition])

Perhaps OAuth2 Flow isn't that dissimilar to "resourceGroup" which is a category, and thus the following could be an alternative:

+ element: category
+ meta
    + classes: flow (array[string], fixed)
+ attributes
	+ grantType
    + scopes

Perhaps `scopes` can be inherited from `OAuth2 Scheme`, the use case of the scopes being different for each flow isn't that clear to me and perhaps we wouldn't do the same in API Blueprint.

So we don't make this a breaking change, we can continue to allow `Member` and `Transition` inside "OAuth2 Scheme", for example:

+ element: OAuth2 Scheme
+ content (Array)
    + Member
    + Transition
    + category|OAuth2 Flow

Thus that ends up being similar to how we can have "groups" of resource groups, or resources inside the API Category.

---

Effectively using the existing terminology the following is the same:

+ element: OAuth2 Scheme
+ attributes
	+ grantType
    + scopes
+ content (Array)
    + Member
    + Transition
    + OAuth2 Scheme

Reply:
Effectively, I am thinking about how we can support what we have already in existing parsers such as OAS 2. While also adding the ability to add multiple flows which are used in OAS 3.

This allows us to add this change as a 1.1 (instead of breaking 2.0) as only the new parser can emit this new structure so it isn't a breaking change which will break existing behaviours for us or any third parties.

That kicks the can down the line of breaking the format, we can alter the OAS 2 parser once we drop the existing structure in API Elements 2.0.